sdks/python/apache_beam/testing/load_tests/pardo_test.py

```diff
@@ -125,7 +125,9 @@ def process(self, element, state=state_param):
             state.add(1)
         yield element
 
-    if self.get_option_or_default('streaming', False):
+    if self.get_option_or_default(
+        'streaming',
+        False) and self.pipeline.get_option('runner') == "PortableRunner":
```

Comment:
The PortableRunner is supposed to use `StatefulLoadGenerator` to enable streaming.
To be honest I assumed that if we are comparing core operations, we should be using the same source method -  SyntheticSources. 
On the other hand maybe `StatefulLoadGenerator` should be part of SyntheticSources. WDYT @tysonjh ?

Reply:
Right, I remember you mentioned there was some issue with the SyntheticSource that Max ran into. Ideally Synthetic source would work for all standard uses but I'm not familiar with the details. How big of a change would it be to fix SyntheticSource?

I'm OK with moving forward with this change, creating a Jira issue for tracking the SyntheticSource improvement, and noting it here as a TODO.

Reply:
Actually, `StatefulLoadGenerator` is no longer required to enable streaming for Flink (please do note I'm using `Flink` instead of `PortableRunner`. `PortableRunner` can run pipelines on any engine that supports portability, Flink included). SyntheticSource executes as SDF (Splittable DoFn). Since Python SDK supports streaming SDF (https://issues.apache.org/jira/browse/BEAM-3742), SyntheticSource should work on Flink.

@mxm We'd love to hear your opinion. Should streaming ParDo tests for Flink still use `StatefulLoadGenerator`?

Reply:
It's definitely not ideal to have two different sources here. Ideally, we would just use `SyntheticSource` for all tests. If we can replicate the behavior of `StatefulLoadGenerator` with `SyntheticSource`, it would be great to remove the former.

Reply:
I did some more tests in streaming on Flink and it turned out that `SyntheticSource` is slower than `StatefulLoadGenerator`. I think it's because `SyntheticSource` is not able to use the benefits of multiple workers (only one task manager generates data).

That being said, I think we should keep two different sources for now. As for the `if` statement above, in my opinion, a new switch (for example, `--use_stateful_load_generator)` be a good idea.